src/frontends/src/query_frontend.hpp

```diff
@@ -386,8 +386,20 @@ class RequestHandler {
     json_response.SetObject();
     clipper::json::add_long(json_response, PREDICTION_RESPONSE_KEY_QUERY_ID,
                             query_response.query_id_);
-    clipper::json::add_double(json_response, PREDICTION_RESPONSE_KEY_OUTPUT,
-                              query_response.output_.y_hat_);
+    try {
+      // Attempt to parse the string output as JSON
+      // and, if possible, nest it in object form within the
+      // query response
+      rapidjson::Document json_y_hat;
+      clipper::json::parse_json(query_response.output_.y_hat_, json_y_hat);
+      clipper::json::add_object(json_response, PREDICTION_RESPONSE_KEY_OUTPUT,
+                                json_y_hat);
+    } catch (const clipper::json::json_parse_error& e) {
+      // If the string output is not JSON-formatted, include
+      // it as a raw string in the query response
+      clipper::json::add_string(json_response, PREDICTION_RESPONSE_KEY_OUTPUT,
+                                query_response.output_.y_hat_);
```

Comment:
I don't think this is safe. If we're including it as a raw string, we need to escape it. Otherwise, a response could mess up our JSON structure. It's not quite this bad (since it won't corrupt our system), but relevant XKCD: https://xkcd.com/327/